### ISSUE TITLE
Selective sampling

### ISSUE THREAD

**Issue by abonander** [chunky-dev/chunky#150]
Let's say I have a large render that's mostly clean, like the following one:

![Noise concentrated in small areas of the image](http://i.imgur.com/25HxzE5.jpg)

Where most of the noise is concentrated in a small area of the image because of the high density of emitters compared to the rest of the scene. It will probably take another 10-40k SPP to clean that noise up. At 1920x1080, that could take a while, considering the given render is currently at 13k SPP after about five hours.

Would it be possible to select only that area of the image to continue rendering? 

If I understand path tracing correctly, the sampling is based on pixels, so it should be as simple as bounding further sampling to the selected area. 

With a reduced sampling area, it should be as clean as the rest of the image in a lot less time than it would have taken sampling the image as a whole, with the render time shortened roughly proportionally by how much smaller the selected area is compared to the canvas.

**Comment by sn0wyfall** [comment]
Yeah I was also thinking of being able to select it myself. I understand that rendering algorithms are tricky. I was also considering literally changing the camera view and selecting rectangular areas with greater noise to render better, but that seems quite complicated haha

**Comment by MadDog443** [comment]
Honestly it might be easier if we had a ![Sky light checkbox](https://user-images.githubusercontent.com/52639083/130314472-a3b381fb-b7ed-410c-a905-b9a1051b2608.png) it would solve a lot of issues cause then we could sample only the emitters which would help a lot, cause we could start out by sampling everything but the emitters then switch to emitters only and set another SPP target.

Easiest way would be to do it the same way the denoiser works, have your skylight renderer, sunlight renderer, and emitter renderer; then you can set your target for each layer to save time rendering.

But if you wanted to go the extra mile, being able to select individual emitters and turn them off would be better, because then we could actually fix problem spots a little easier.

**Comment by Peregrine05** [comment]
I believe that lens shift would be a somewhat complicated way to achieve this result, by narrowing the canvas to be one specific area of the larger canvas. Then only the pixels in the new smaller canvas are being sampled.
